lib/sidekiq/cloudwatchmetrics.rb

```diff
@@ -34,7 +36,7 @@ def self.enable!(**kwargs)
   end
 
   class Publisher
-    include Sidekiq::Util
+    SIDEKIQ_GTE_6_5_0 ? (include Sidekiq::Component) : (include Sidekiq::Util)
```

Comment:
⚠️ Sidekiq 6.5's `Sidekiq::Component` relies on an attribute reader `config` to be set (see https://github.com/mperham/sidekiq/blob/main/lib/sidekiq/component.rb#L3-L5).

Otherwise `safe_thread` will produce a `NoMethodError` when rescuing an exception (which calls `handle_exception`.

`Component` also delegates `logger` / `redis` to this `config` attribute - which is why I'm guessing you updated this to use `Sidekiq.logger` everywhere it was using `logger`.

I suggest adding a line in the initializer to cover that:
```ruby
@config = Sidekiq
```

Then you can leave the logger lines as `logger.info` without specifying `Sidekiq`

Reply:
Thanks for picking that up. I've made the suggested change :slightly_smiling_face: